packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -566,9 +595,26 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
       return matrix.clone();
     }
 
-    final Offset alignedTranslation = widget.alignPanAxis && _panAxis != null
-      ? _alignAxis(translation, _panAxis!)
-      : translation;
+    late final Offset alignedTranslation;
+
+    if (_panAxis != null) {
+      switch(widget.panAxis){
+        case PanAxis.horizontal:
+          alignedTranslation = _alignAxis(translation, Axis.horizontal);
+          break;
+        case PanAxis.vertical:
+          alignedTranslation = _alignAxis(translation, Axis.vertical);
+          break;
+        case PanAxis.aligned:
+          alignedTranslation = _alignAxis(translation, _panAxis!);
```

Comment:
Can you update the comment at the definition of _panAxis?  It still references alignPanAxis.

Also it's kind of confusing that there is both `panAxis` and `_panAxis`, but they mean different things and have different types...  Maybe if we just change _panAxis to _panningAxis or _currentAxis or something?  Up to you.

Reply:
All done!